packages/plugins/users-permissions/server/controllers/auth.js

```diff
@@ -273,20 +273,13 @@ module.exports = {
       throw new ApplicationError('Register action is currently disabled');
     }
 
+    const { register } = strapi.config.get('plugin.users-permissions');
+
     const params = {
-      ..._.omit(ctx.request.body, [
-        'confirmed',
-        'blocked',
-        'confirmationToken',
-        'resetPasswordToken',
-        'provider',
-        'id',
-        'createdAt',
-        'updatedAt',
-        'createdBy',
-        'updatedBy',
-        'role',
-      ]),
+      ..._.pick(
+        ctx.request.body,
+        register?.allowedFields ? register?.allowedFields : ['username', 'email', 'password']
```

Comment:
I wouldn't allow overriding the three required fields, we should make it "additionalFields" and merge it with username, email, password.

Reply:
Agree with Ben here